tests/unit/amo/components/TestAddon.js

```diff
@@ -96,6 +123,72 @@ describe('Addon', () => {
     expect(rootNode.querySelector('h1').textContent).toContain('Chill Out');
   });
 
+  it('renders without an add-on', () => {
+    const errorHandler = new ErrorHandler({
+      id: 'no-addon-error-handler',
+      dispatch: sinon.stub(),
+    });
+    const slugParam = 'some-addon'; // as passed through the URL.
+    const fakeDispatch = sinon.stub();
+
+    // Simulate the case when an add-on has not been loaded into state yet.
+    const root = shallowRender({
+      addon: null,
+      errorHandler,
+      dispatch: fakeDispatch,
+      params: { slug: slugParam },
+    });
+
+    // Since there's no add-on, it should be fetched on load.
+    sinon.assert.calledWith(
+      fakeDispatch, fetchAddonAction({ errorHandler, slug: slugParam }));
+
+    // These should be empty:
+    expect(root.find(InstallButton)).toHaveLength(0);
+    expect(root.find(AddonCompatibilityError)).toHaveLength(0);
```

Comment:
I do like that `toHaveLength(0)` and `toHaveLength(1)` contrast well with each other in this test though, so that's a reason to keep using it over the `toBeEmpty()` matcher.